aesd-char-driver/main.c

```diff
@@ -162,12 +165,95 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
         return retval;
 }
 
+loff_t aesd_llseek(struct file *filp, loff_t off, int whence)
+{
+    int idx = 0;
+    loff_t retval = 0;
+    loff_t total_buf_size = 0;
+    struct aesd_dev *dev = filp->private_data;
+
+    // check input parameters
+    if (filp == NULL || dev == NULL) 
+        return -EINVAL;
+
+    // acquire lock 
+    if (mutex_lock_interruptible(&dev->lock)) 
+        return -ERESTARTSYS;
+
+    struct aesd_buffer_entry *entry = NULL;
+    AESD_CIRCULAR_BUFFER_FOREACH(entry, &dev->buffer, idx) {
+       total_buf_size += entry->size;
+    }
+
+    // calling fixed llsek for location of offset
+    retval = fixed_size_llseek(filp, off, whence, total_buf_size);
```

Comment:
@KapureCUB no need to lock from here through the end of the function since dev is not accessed.